sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -39,7 +39,9 @@
 #endif
 
 #ifndef __SYCL_DEPRECATED
-#ifndef SYCL_DISABLE_DEPRECATION_WARNINGS
+// The deprecated attribute is not supported in some situations(e.g. namespace)
+// in C++14 mode
+#if !defined(SYCL2020_DISABLE_DEPRECATION_WARNINGS) && __cplusplus >= 201703L
```

Comment:
Typically you're better off checking the feature-test-macro.  That way you can get support for the features you want in earlier compilers.

Reply:
Ok, will try to find corresponding feature-test-macro.

Reply:
I've been looking for a paper that adds namespace support to deprecated, and I just don't see it.I found a GCC bug that added it (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79817), but they don't reference any standards paper.  @AaronBallman

Reply:
It's a C++17 feature that was added by http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4196.html

Reply:
Ah, i see!  Sadly it doesn't look like it was given a feature-test-macro.

Reply:
Yeah, it looks like no one proposed to change the value from `__has_cpp_attribute()` when the change was adopted. :-(